File: php/src/Resource/NotificationChannels.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Resource;

use Fivexer\SDK\Fivexer;
use Fivexer\SDK\Internal\Json;
use Fivexer\SDK\Model\CreateNotificationChannel;
use Fivexer\SDK\Model\NotificationChannel;
use Fivexer\SDK\Model\UpdateNotificationChannel;

/** Channels describe where to deliver. The signing secret is write-only and never returned. */
final class NotificationChannels
{
    public function __construct(private readonly Fivexer $client)
    {
    }

    /** @return list<NotificationChannel> */
    public function list(): array
    {
        $data = $this->client->request('GET', '/notification-channels') ?? [];
        /** @var list<NotificationChannel> */
        return Json::parseEach($data, 'channels', [NotificationChannel::class, 'fromArray']);
    }

    public function create(CreateNotificationChannel $input): NotificationChannel
    {
        return NotificationChannel::fromArray(
            $this->client->request('POST', '/notification-channels', $input->toArray()) ?? []
        );
    }

    public function get(string $channelId): NotificationChannel
    {
        return NotificationChannel::fromArray(
            $this->client->request('GET', '/notification-channels/' . \rawurlencode($channelId)) ?? []
        );
    }

    /** Partial update — fields left unset keep their stored value. */
    public function update(string $channelId, UpdateNotificationChannel $update): NotificationChannel
    {
        return NotificationChannel::fromArray(
            $this->client->request(
                'PATCH',
                '/notification-channels/' . \rawurlencode($channelId),
                $update->toArray()
            ) ?? []
        );
    }

    public function remove(string $channelId): void
    {
        $this->client->request('DELETE', '/notification-channels/' . \rawurlencode($channelId));
    }
}

File: java/src/main/java/io/fivexer/sdk/NotificationChannels.java
package io.fivexer.sdk;

import io.fivexer.sdk.internal.Json;
import io.fivexer.sdk.model.CreateNotificationChannel;
import io.fivexer.sdk.model.NotificationChannel;
import io.fivexer.sdk.model.NotificationChannelList;
import io.fivexer.sdk.model.UpdateNotificationChannel;
import java.util.Collections;
import java.util.List;

/** Channels describe where to deliver. The signing secret is write-only and never returned. */
public final class NotificationChannels {

    private final Fivexer client;

    NotificationChannels(Fivexer client) {
        this.client = client;
    }

    public List<NotificationChannel> list() {
        NotificationChannelList envelope = client.request("GET", "/notification-channels", null, null,
                NotificationChannelList.class, false);
        return envelope.getChannels() == null ? Collections.emptyList() : envelope.getChannels();
    }

    public NotificationChannel create(CreateNotificationChannel input) {
        return client.request("POST", "/notification-channels", input.toJson(), null,
                NotificationChannel.class, false);
    }

    public NotificationChannel get(String channelId) {
        return client.request("GET", "/notification-channels/" + Json.enc(channelId), null, null,
                NotificationChannel.class, false);
    }

    /** Partial update — fields left unset keep their stored value. */
    public NotificationChannel update(String channelId, UpdateNotificationChannel update) {
        return client.request("PATCH", "/notification-channels/" + Json.enc(channelId), update.toJson(),
                null, NotificationChannel.class, false);
    }

    public void remove(String channelId) {
        client.request("DELETE", "/notification-channels/" + Json.enc(channelId), null, null,
                Void.class, true);
    }
}
